src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,25 @@
+package duke;
+
+import duke.command.Command;
+
+public class Duke {
+
+    public static void main(String[] args) {
+        Storage storage = new Storage();
+        TaskList list = storage.readTaskList();
+        UI.greet();
+        do {
+            try {
+                String input = UI.readCommand();
+                Command cmd = Parser.parseCommand(input);
+                cmd.execute(list);
+                if (cmd.isExit()) {
```

Comment:
Perhaps you can incorporate this condition into your while loop to avoid nesting?